src/users/users.controller.ts

```diff
@@ -234,6 +238,60 @@ export class UsersController {
     }
   }
 
+  @Delete('/one/:id/photo')
+  @UseGuards(AuthGuard)
+  @JwtType(JwtProcessorType.RSA)
+  @ApiOperation({
+    description: SWAGGER_DESC_DELETE_PHOTO_USER_BY_ID,
+  })
+  @ApiOkResponse({
+    description: 'Deletes user profile photo',
+  })
+  @ApiNoContentResponse({
+    description: 'Returns empty content if there was no user profile photo',
+  })
+  @ApiForbiddenResponse({
+    description: 'Returns when user is not authenticated',
+  })
+  @ApiUnauthorizedResponse({
+    description: 'Returns when isAdmin is false',
+  })
+  async deleteUserPhotoById(
+    @Param('id') id: number,
+    @Query('isAdmin') isAdminParam: string,
+    @Res({ passthrough: true }) res: FastifyReply,
+  ) {
+    isAdminParam = isAdminParam.toLowerCase();
+    const isAdmin =
+      isAdminParam === 'true' || isAdminParam === '1' ? true : false;
+    if (!isAdmin) {
+      throw new UnauthorizedException();
+    }
+
+    const user = await this.usersService.findById(id);
+    if (!user) {
+      throw new NotFoundException({
+        error: 'Could not file user',
+        location: __filename,
+      });
+    }
+
+    if (!user.photo) {
+      res.status(HttpStatus.NO_CONTENT);
+      return;
+    }
+
+    try {
+      this.usersService.deletePhoto(id);
+      return;
+    } catch (err) {
+      throw new InternalServerErrorException({
+        error: err.message,
+        location: __filename,
+      });
+    }
```

Comment:
Please remove the try/catch block as it will suppress the original error raised from the `deletePhoto` method, such as a `NotFoundError`.